docs/src/theory.md

```diff
@@ -0,0 +1,314 @@
+
+# Theory of Bi-level Optimization
+## Introduction
+Standard single-level optimization problems consist of a single decision-maker who controls a set of decision variables to optimize its goal. However, in many real-world problems, several decision-makers with conflicting goals interact, and each individual decision impacts the decisions and outcomes of the others. Such interdependencies cannot be captured with the standard-setting of a single decision-maker, and other, more adequate, optimization problems are required [[1]](#1).
```

Comment:
I'm not sure we should copy verbatim the first paragraph of Kleinert's thesis, even if it is cited.

Reply:
agreed, things should be much shorter, more informal in the documentation